File: handle_csv.dart
List active = [];

List activities(var data2) {
  var data = "";
  for (var row in data2) {
    if (row[0].contains(",")) {
      data = row[0].split(",")[0];
    } else {
      data = row[0];
    }
    if (!active.contains(data)) {
      active.add(data);
    }
  }
  return active;
}

List getDur(var dat, var csv) {
  List cond = [];
  var name;
  var data;
  for (var row in csv) {
    if (row[0].contains(",")) {
      name = row[0].split(",")[0];
    } else {
      name = row[0];
    }
    if (name == dat) {
      if (row[0].contains(",")) {
        data = row[0].split(",")[1];
      } else {
        if (num.tryParse(row[1].toString()) != null) {
          data = "";
        } else {
          data = row[1];
        }
      }
      if (!cond.contains(data)) {
        cond.add(data);
      }
    }
  }
  return cond;
}

double burned(var activity, var state, var csv) {
  var name;
  var data;
  var burned;
  for (var row in csv) {
    if (row[0].contains(",")) {
      name = row[0].split(",")[0];
    } else {
      name = row[0];
    }
    if (name == activity) {
      if (row[0].contains(",")) {
        data = row[0].split(",")[1];
      } else {
        if (num.tryParse(row[1].toString()) != null) {
          data = "";
        } else {
          data = row[1];
        }
      }
      if (data == state) {
        return row[row.length - 1];
      }
    }
  }
  return burned;
}

File: handle_csv.py
import csv

activities = []
cond = []


def extract_activ():
    with open('exercise_dataset.csv', newline='') as csvfile:
        reader = csv.reader(csvfile, delimiter=',')
        data = ""
        for row in reader:
            if "," in row[0]:
                data = row[0].split(",")[0]
            else:
                data = row[0]
            if data in activities:
                pass
            else:
                activities.append(data)

    return activities


def cs2(dat):
    with open('exercise_dataset.csv', newline='') as csvfile:
        reader = csv.reader(csvfile, delimiter=',')
        name = ""
        data = ""
        for row in reader:
            if "," in row[0]:
                name = row[0].split(",")[0]
            else:
                name = row[0]
            if name == dat:
                if "," in row[0]:
                    data = row[0].split(",")[1]
                else:
                    if row[1].isdigit():
                        data = ""
                    else:
                        data = row[1]
                if data not in cond:
                    cond.append(data)
    return cond


